core/Execute.cpp

```diff
@@ -38,10 +39,83 @@ namespace olympia
                                                                         &exe_pipe_fact_));
             }
         }
+
+        issue_queue_topology_ = olympia::coreutils::getPipeTopology(node->getParent(), "issue_queue_topology");
+        for(size_t i = 0; i < issue_queue_topology_.size(); ++i){
+            // loop through the issue_queue_topology and create a corresponding issue_queue sparta unit
+            std::string exe_units_string = "";
+            for(auto exe_unit: issue_queue_topology_[i]){
+                exe_units_string += exe_unit + ", ";
+            }
+            const std::string issue_queue_name = "iq" + std::to_string(i);
+            // naming it for the target type it supports, i.e issue_queue_alu or issue_queue_fpu
+            const std::string tgt_name = "IssueQueue_" + issue_queue_topology_[i][0].substr(0, issue_queue_topology_[i][0].size()-1);
+            //const std::string tgt_name = issue_queue_topology_[i][0].substr(0, issue_queue_topology_[i][0].size()-1);
+            auto iq_node = new sparta::ResourceTreeNode(node,
+                                                        issue_queue_name,
+                                                        tgt_name,
+                                                        i,
+                                                        exe_units_string + std::string("Issue Queue"),
+                                                        &issue_queue_fact_);
+            // auto iq_resource = iq_node->getResourceAs<olympia::IssueQueue*>();
+            issue_queues_.emplace_back(iq_node);
+        }
+
     }
+    void ExecuteFactory::bindLate(sparta::TreeNode* node){
+        /*
+            For issue queue we need to establish mappings such that a mapping of target pipe to execution pipe
+            in an issue queue is known such as:
+                iq_0:
+                    "int": alu0, alu1
+                    "div": alu1
+                    "mul": alu3
+            so when we have an instruction, we can get the target pipe of an instruction and lookup available
+            execution units
+        */
+        std::unordered_map<std::string, int> pipe_to_iq_number;
+
+        for (size_t i = 0; i < issue_queues_.size(); ++i) {
+            for (auto alu_name : issue_queue_topology_[i]) {
+                for (auto& exe_pipe_tns : exe_pipe_tns_) {
+                    olympia::ExecutePipe* exe_pipe = exe_pipe_tns->getResourceAs<olympia::ExecutePipe*>();
+                    std::string exe_unit_name = exe_pipe->getName();
+
+                    if (alu_name == exe_unit_name) {
```

Comment:
Not sure what this code is doing.  Can you explain why the alu is explicitly called out?

Reply:
Changed the variable name to exe_name, we are searching for the corresponding execution_pipe for a given exe_unit defined in the .yaml and compare based on the name i.e `alu0` to build a map for later use in Execute.cpp and for use in IssueQueue for setting which executepipes belong to that issue queue.